test/util/timeout-registrySpec.ts

```diff
@@ -70,6 +71,10 @@ describe('timeout registry', () => {
         })
 
         it('clears timeout when connection lost', done => {
+            services.loggerMock
+                .expects('warn')
+                .never()
+
```

Comment:
You still need to test if the timeoutRegistry has called the timeRegistry.remove service for each timeout, no?

Reply:
I agree, it should actually assert for that, however for now `onConnectionLost` clears timeouts using `clearTimeout` without relying on the `timerRegistry` service. I wonder whether `onConnectionLost` should be refactored to use the services instead?